src/OpenIddict.Client/OpenIddictClientOptions.cs

```diff
@@ -157,4 +157,11 @@ public sealed class OpenIddictClientOptions
     /// </summary>
     [EditorBrowsable(EditorBrowsableState.Advanced)]
     public HashSet<string> ResponseTypes { get; } = new(StringComparer.Ordinal);
+
+#if SUPPORTS_TIME_PROVIDER
+    /// <summary>
+    /// Gets the time provider
+    /// </summary>
+    public TimeProvider? TimeProvider { get; set; }
```

Comment:
Or maybe we should leave it nullable and do `TimeProvider?.GetUtcNow() ?? DateTimeOffset.UtcNow` where it's used? It would address the issue you had in the unit tests.